spanner/changestreams/src/main/java/com/example/spanner/changestreams/ChangeStreamSample.java

```diff
@@ -0,0 +1,236 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.spanner.changestreams;
+
+// TODO(haikuo@google.com): we should remove these models once the change stream connector code that
+// includes these models are released into Beam repo.
+import com.example.spanner.changestreams.model.ChangeStreamRecord;
+import com.example.spanner.changestreams.model.ChildPartition;
+import com.example.spanner.changestreams.model.ChildPartitionsRecord;
+import com.example.spanner.changestreams.model.DataChangeRecord;
+import com.example.spanner.changestreams.model.HeartbeatRecord;
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.cloud.Timestamp;
+import com.google.cloud.spanner.DatabaseAdminClient;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.Statement;
+import com.google.common.collect.ImmutableList;
+import com.google.spanner.admin.database.v1.UpdateDatabaseDdlMetadata;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Sample code for querying change stream, it:
+ * 1. Creates a table with simple schema and a change stream that watches the table.
+ * 2. Inserts test data into the table.
+ * 3. Executes a change stream initial query to get change stream partition tokens.
+ * 4. Executes a change stream partition query to get data change records of the inserted rows.
+ * 5. Drops the created table and change stream.
+ */
+public class ChangeStreamSample {
+  private static final long TIMEOUT_MINUTES = 2;
+  private static final ChangeStreamRecordMapper changeStreamRecordMapper =
+      new ChangeStreamRecordMapper();
+
+  public static void run(String instanceId, String databaseId, String prefix) {
+    final String tableName = prefix + "_Singers";
+    final String changeStreamName = prefix + "_ChangeStreamSingers";
+    final SpannerOptions options = SpannerOptions.newBuilder().build();
+    final Spanner spanner = options.getService();
+    final DatabaseAdminClient dbAdminClient = spanner.getDatabaseAdminClient();
+    final DatabaseClient dbClient = spanner.getDatabaseClient(
+        DatabaseId.of(options.getProjectId(), instanceId, databaseId));
+
+    try {
+      createChangeStream(dbAdminClient, instanceId, databaseId, tableName, changeStreamName);
+      queryChangeStream(dbClient, tableName, changeStreamName);
+      dropChangeStream(dbAdminClient, instanceId, databaseId, tableName, changeStreamName);
+    } catch (Exception e) {
+      System.out.println(e);
+    } finally {
+      spanner.close();
+    }
+  }
+
+  // [START change_stream_sample_create_change_stream]
```

Comment:
Update the tags to contain `spanner` (as in `spanner_change_stream_sample_create_change_stream`). This should fix the snippet-bot failure you are getting

Reply:
Btw, I think we should use change_streams (plural) instead of change_stream (singular)

Reply:
Changed, let me know if you prefer to change all the naming to plural, I didn't since it's a bit wired to say "read a change streams", and a lots of our code is already using change stream, "change streams" seems like something makes more sense in terms marking, I am fine either ways though.